src/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs

```diff
@@ -179,6 +182,16 @@ private static IEnumerable<object[]> ProtocolMismatchData()
             return true;  // allow everything
         }
 
+        // The following method is invoked by the RemoteCertificateValidationDelegate.
+        private bool AllowAnyServerCertificate(
+              object sender,
+              X509Certificate certificate,
+              X509Chain chain,
+              SslPolicyErrors sslPolicyErrors)
+        {
```

Comment:
I don't understand this change.  This effectively moved the assert that was for the server side to now be on the client side.  I would revert this change and just use `delegate { return true; }` when constructing the client SslStream.

Reply:
It does not actually move the assert, only the names are flipped.

However, it might really be cleaner to use the delegate that you suggested on the client side and rename the server into `AllowEmptyClientCertificate` - because what's really happening is that the server side's validation function is called with a `null` client certificate.  How does that sound?

Reply:
> It does not actually move the assert, only the names are flipped.

Ah, because you not only swapped the names, you also swapped the call sites.  Ok.

> However, it might really be cleaner to use the delegate that you suggested on the client side

Sure, sounds good.